### PR TITLE
[tests] Rework how we AOT compile tests + General cleanup of make targets

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

File: mono/tests/Makefile.am [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+	$(MAKE) test-console-output
+	$(MAKE) test-env-options
+	$(MAKE) test-unhandled-exception-2
+if !FULL_AOT_TESTS
+	$(MAKE) test-appdomain-unload
 endif
+	$(MAKE) test-process-stress
+	$(MAKE) test-pedump
+	$(MAKE) rm-empty-logs
+	$(MAKE) runtest-gac-loading
```
Comment (vargaz) [inline mono/tests/Makefile.am]: What is the purpose of this change ? Why can't we use normal make dependencies ?

Comment (luhenry) [inline mono/tests/Makefile.am]: So if we do `make -j4 check-local ` we don't run them in parallel.

Comment (vargaz) [inline mono/tests/Makefile.am]: Makes sense, the output did look a bit hard to read.

**Comment by lewurm** [comment]
@monojenkins build interpreter

**Comment by luhenry** [comment]
@monojenkins build interpreter
